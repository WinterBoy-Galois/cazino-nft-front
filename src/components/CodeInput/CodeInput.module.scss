@import '../../styles/shared';

.container {
  input {
    border-radius: 10px;
    font-size: 28px;
    text-align: center;
    margin-right: 0.5rem;
    border: 3px solid transparent;
    line-height: 1;
    width: 40px;

    @supports (-webkit-touch-callout: none) {
      width: auto;
    }

    @include media-breakpoint-up(md) {
      font-size: 50px;
      width: 56px;
    }

    &:last-of-type {
      margin-right: 0;
    }

    &:focus {
      border: 3px solid $activation-code-border-blue;
    }
  }

  input[type='number'] {
    -moz-appearance: textfield;
  }

  /* stylelint-disable */
  input::-webkit-outer-spin-button,
  input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
  /* stylelint-enable */
}
