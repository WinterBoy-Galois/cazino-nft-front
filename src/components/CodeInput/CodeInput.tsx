import React from 'react';
import styles from './CodeInput.module.scss';
import ReactCodeInput from 'react-code-input';

interface IProps {
  name?: string;
  fields?: number;
}

const CodeInput: React.SFC<IProps> = ({ name = 'code-input', fields = 6 }) => {
  return (
    <ReactCodeInput
      name={name}
      inputMode={'numeric'}
      type={'number'}
      fields={fields}
      className={styles.container}
      filterKeyCodes={[188, 187, 190, 189, 109, 107, 110]}
    />
  );
};

export default CodeInput;
