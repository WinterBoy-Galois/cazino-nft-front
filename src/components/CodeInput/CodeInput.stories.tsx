import React from 'react';
import { number } from '@storybook/addon-knobs';

import CodeInput from '.';

export default {
  title: 'Components/CodeInput',
  component: CodeInput,
};

const data = {
  fields: 6,
};

export const Default = () => <CodeInput fields={number('Fields', data.fields)} />;
