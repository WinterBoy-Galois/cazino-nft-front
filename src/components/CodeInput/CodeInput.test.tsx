import React from 'react';
import { render } from '@testing-library/react';

import CodeInput from '.';

describe('CodeInput', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<CodeInput message="Hello World!" />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
