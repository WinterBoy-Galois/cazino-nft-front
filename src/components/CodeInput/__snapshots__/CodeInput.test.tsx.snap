// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeInput should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container react-code-input"
        style="display: inline-block;"
      >
        <input
          autocomplete="off"
          data-id="0"
          data-valid="true"
          id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-0"
          inputmode="numeric"
          max="9"
          maxlength="6"
          min="0"
          type="number"
          value=""
        />
        <input
          autocomplete="off"
          data-id="1"
          data-valid="true"
          id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-1"
          inputmode="numeric"
          max="9"
          maxlength="6"
          min="0"
          type="number"
          value=""
        />
        <input
          autocomplete="off"
          data-id="2"
          data-valid="true"
          id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-2"
          inputmode="numeric"
          max="9"
          maxlength="6"
          min="0"
          type="number"
          value=""
        />
        <input
          autocomplete="off"
          data-id="3"
          data-valid="true"
          id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-3"
          inputmode="numeric"
          max="9"
          maxlength="6"
          min="0"
          type="number"
          value=""
        />
        <input
          autocomplete="off"
          data-id="4"
          data-valid="true"
          id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-4"
          inputmode="numeric"
          max="9"
          maxlength="6"
          min="0"
          type="number"
          value=""
        />
        <input
          autocomplete="off"
          data-id="5"
          data-valid="true"
          id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-5"
          inputmode="numeric"
          max="9"
          maxlength="1"
          min="0"
          type="number"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container react-code-input"
      style="display: inline-block;"
    >
      <input
        autocomplete="off"
        data-id="0"
        data-valid="true"
        id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-0"
        inputmode="numeric"
        max="9"
        maxlength="6"
        min="0"
        type="number"
        value=""
      />
      <input
        autocomplete="off"
        data-id="1"
        data-valid="true"
        id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-1"
        inputmode="numeric"
        max="9"
        maxlength="6"
        min="0"
        type="number"
        value=""
      />
      <input
        autocomplete="off"
        data-id="2"
        data-valid="true"
        id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-2"
        inputmode="numeric"
        max="9"
        maxlength="6"
        min="0"
        type="number"
        value=""
      />
      <input
        autocomplete="off"
        data-id="3"
        data-valid="true"
        id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-3"
        inputmode="numeric"
        max="9"
        maxlength="6"
        min="0"
        type="number"
        value=""
      />
      <input
        autocomplete="off"
        data-id="4"
        data-valid="true"
        id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-4"
        inputmode="numeric"
        max="9"
        maxlength="6"
        min="0"
        type="number"
        value=""
      />
      <input
        autocomplete="off"
        data-id="5"
        data-valid="true"
        id="0b57f882-db68-41a1-b9e2-f1b5a4bd9541-5"
        inputmode="numeric"
        max="9"
        maxlength="1"
        min="0"
        type="number"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
